File: src/main/java/com/vicen/films/application/service/FilmService.java
package com.vicen.films.application.service;

import com.vicen.films.application.commands.CreateFilmCommand;
import com.vicen.films.application.queries.GetFilmsQuery;
import com.vicen.films.domain.films.Film;

import java.util.List;

public interface FilmService {

    Film createFilm(CreateFilmCommand command);

    List<Film> getFilms(GetFilmsQuery query);
}

File: src/main/kotlin/com/vicen/films/application/service/FilmService.kt
package com.vicen.films.application.service

import com.vicen.films.application.commands.CreateFilmCommand
import com.vicen.films.application.queries.GetFilmQuery
import com.vicen.films.application.queries.GetFilmsQuery
import com.vicen.films.domain.films.Film

interface FilmService {

    fun getFilms(query: GetFilmsQuery = GetFilmsQuery()): List<Film>
    fun getFilm(query: GetFilmQuery): Film
    fun createFilm(command: CreateFilmCommand): Film

}